packages/server/src/course/course.controller.ts

```diff
@@ -18,52 +18,54 @@ import { User } from '../profile/user.decorator';
 import { UserModel } from '../profile/user.entity';
 import { QueueModel } from '../queue/queue.entity';
 import { CourseModel } from './course.entity';
+import { QueueService } from '../queue/queue.service';
 
 @Controller('courses')
 @UseGuards(JwtAuthGuard)
 @UseInterceptors(ClassSerializerInterceptor)
 export class CourseController {
-  constructor(private connection: Connection) {}
+  constructor(
+    private connection: Connection,
+    private queueService: QueueService,
+  ) {}
 
   @Get(':id')
   async get(@Param('id') id: number): Promise<GetCourseResponse> {
     // TODO: for all course endpoint, check if they're a student or a TA
     const course = await CourseModel.findOne(id, {
-      relations: ['officeHours'],
+      relations: [
+        'officeHours',
+        'queues',
+        'queues.staffList',
+        'queues.officeHours',
+      ],
     });
 
     const now = new Date();
     const MS_IN_MINUTE = 60000;
 
-    const officeHoursHappeningNow = course.officeHours.filter(
-      (e) =>
-        e.startTime.valueOf() - 15 * MS_IN_MINUTE < now.valueOf() &&
-        e.endTime.valueOf() + 1 * MS_IN_MINUTE > now.valueOf(),
-    );
-
-    const queues = await QueueModel.find({
-      where: {
-        courseId: id,
-      },
-      relations: ['staffList'],
-    });
-
-    const nonEmptyQueues = queues.filter((e) => e.staffList.length > 0);
-
-    const queuesHappeningNow = [];
+    // Set all the queues which should be "Open"
+    course.queues = course.queues.filter((queue) => {
+      const areStaffPresent = queue.staffList.length > 0;
```

Comment:
should we just filter on `queue.isOpen`?

Reply:
yeah good call. Just fixed that up